Updated `Lightning service provider` page (#809)
BitcoinDesign/Guide: guide/how-it-works/lightning-services.md

---
layout: guide
title: Lightning Services
description: Third party services that solve common user experience problems when connecting to and using the lightning network.
nav_order: 4
parent: How it works
permalink: /guide/how-it-works/lightning-services/
redirect_from:
 - /guide/payments/send/lightning-service-providers/
main_classes: -no-top-padding
---

<!--

Editor's notes

Illustration sources: https://www.figma.com/file/qr4P17z6WSPADm6oW0cKw2/?node-id=701%3A5911

-->

{% include picture.html
   image = "/assets/images/guide/how-it-works/lightning-services/lightning-services.jpg"
   retina = "/assets/images/guide/how-it-works/lightning-services/lightning-services@2x.jpg"
   mobile = "/assets/images/guide/how-it-works/lightning-services/lightning-services-mobile.jpg"
   mobileRetina = "/assets/images/guide/how-it-works/lightning-services/lightning-services-mobile@2x.jpg"
   alt-text = "Lightning service providers header image."
   caption = "Image by [Rahul Viswanath](https://unsplash.com/@rahul_viswanath) from Unsplash"
   width = 1600
   height = 740
   layout = "full-width"
%}

# Lightning services

> [Scaling]({{ '/guide/getting-started/technology-primer/#do-all-transactions-have-to-be-this-secure' | relative_url }}) bitcoin to billions of people non-custodially with the lightning network creates many new usability issues. Including liquidity and uptime requirements, channel management, routing, and more complex backups. Lightning services offered by third-parties aim to solve these issues.

Reviewer: An attempt to make things easier to "scan". 
```suggestion
[Scaling]({{ '/guide/getting-started/technology-primer/#do-all-transactions-have-to-be-this-secure' | relative_url }}) bitcoin to billions of people non-custodially with the lightning network creates many new usability issues. 
Some of those issues include:
- liquidity and uptime requirements
- channel management
- routing 
- complex backups
Lightning services offered by third-parties aim to solve these issues.
```

Author: Good suggestion!

> In this section, we cover two categories of lightning services. Lightning service providers (LSPs), which focus on onboarding and providing connectivity to lightning users, and [Lightning wallet servers]({{ '/guide/how-it-works/lightning-services/#what-are-lightning-wallet-servers' | relative_url }}), which provide additional services not relating to connectivity.

Reviewer: consistency in lowercase
```suggestion
In this section, we cover two categories of lightning services. Lightning service providers (LSPs), which focus on onboarding and providing connectivity to lightning users, and [lightning wallet servers]({{ '/guide/how-it-works/lightning-services/#what-are-lightning-wallet-servers' | relative_url }}), which provide additional services not relating to connectivity.
```

> ## What is a Lightning service provider?

Reviewer: lowercase consistency
```suggestion
## What is a lightning service provider?
```

Author: Not sure about this one as an LSP, like an ISP, is a proper name

Reviewer: Not a blocker for me, as you say.

Author: Lighting service provider and lightning (which I lowercase) are two separate things

---
layout: guide
title: Lightning Services
description: Third party services that solve common user experience problems when connecting to and using the lightning network.
nav_order: 4
parent: How it works
permalink: /guide/how-it-works/lightning-services/
redirect_from:
 - /guide/payments/send/lightning-service-providers/
main_classes: -no-top-padding

> ---

Author: I had one added, did it not show up?

> ---

Reviewer 2: I don't see the link preview image anywhere in lines 1 - 11. And no, it's not showing up when doing an OpenGraph data fetch.

<!--

Editor's notes

Illustration sources: https://www.figma.com/file/qr4P17z6WSPADm6oW0cKw2/?node-id=701%3A5911

-->

{% include picture.html
   image = "/assets/images/guide/how-it-works/lightning-services/lightning-services.jpg"
   retina = "/assets/images/guide/how-it-works/lightning-services/lightning-services@2x.jpg"
   mobile = "/assets/images/guide/how-it-works/lightning-services/lightning-services-mobile.jpg"
   mobileRetina = "/assets/images/guide/how-it-works/lightning-services/lightning-services-mobile@2x.jpg"
   alt-text = "Lightning service providers header image."
   caption = "Image by [Rahul Viswanath](https://unsplash.com/@rahul_viswanath) from Unsplash"
   width = 1600
   height = 740
   layout = "full-width"
%}

# Lightning services

> [Scaling]({{ '/guide/getting-started/technology-primer/#do-all-transactions-have-to-be-this-secure' | relative_url }}) bitcoin to billions of people non-custodially with the lightning network creates many new usability issues. 

Reviewer 2: ```suggestion
[Scaling]({{ '/guide/getting-started/technology-primer/#do-all-transactions-have-to-be-this-secure' | relative_url }}) bitcoin to billions of self-custodial users with the lightning network creates many new usability issues. 
```

Some of those issues include:
- liquidity and uptime requirements
- channel management
- routing 
- complex backups
Lightning services offered by third-parties aim to solve these issues.

In this section, we cover two categories of lightning services. Lightning service providers (LSPs), which focus on onboarding and providing connectivity to lightning users, and [lightning wallet servers]({{ '/guide/how-it-works/lightning-services/#what-are-lightning-wallet-servers' | relative_url }}), which provide additional services not relating to connectivity.

## What is a Lightning service provider?

> Similar to how Internet service providers (ISPs) help users connect to and use the internet, Lightning service providers (LSPs) help users connect to and use lightning [non-custodially]({{ '/guide/getting-started/principles/#self-custody' | relative_url }}).

Author: Wikipedia states it like so but yeah dictionary definitions just refer to connecting, will go with this wording

They do this by being well connected in the network, opening channels, and offering inbound liquidity to users. They often charge a fee for their services.

You can learn more about how liquidity works on our [liquidity]({{ '/guide/how-it-works/liquidity/' | relative_url }}) page.

Some LSPs may also offer channel management, routing, backups and [other services]({{ '/guide/how-it-works/lightning-services/#what-are-lightning-wallet-servers' | relative_url }}), but these are not limited to being offered by LSPs.

{% include picture.html
   image = "/assets/images/guide/how-it-works/lightning-services/Inbound-liquidity-as-a-service.png"
   retina = "/assets/images/guide/how-it-works/lightning-services/Inbound-liquidity-as-a-service@2x.png"
   mobile = "/assets/images/guide/how-it-works/lightning-services/Inbound-liquidity-as-a-service-mobile.png"
   mobileRetina = "/assets/images/guide/how-it-works/lightning-services/Inbound-liquidity-as-a-service-mobile@2x.png"
   alt-text = "Graphic depicting how LSPs offer liquidity as a service"
   width = 800
   height = 348
%}

## How do LSPs connect users to lightning?

> To connect to and use the lightning non-custodially, a user needs a [payment channel]({{ '/guide/getting-started/technology-primer/#what-is-a-payment-channel' | relative_url }}). To send payments, this channel needs outbound liquidity, which the user usually provides. And to receive payments, this channel needs inbound liquidity, which needs to be provided by someone else.

Reviewer 2: ```suggestion
To use lightning, a self-custodial user needs at least one  [payment channel]({{ '/guide/getting-started/technology-primer/#what-is-a-payment-channel' | relative_url }}). To send payments, this channel needs outbound liquidity, which the user usually provides. And to receive payments, this channel needs inbound liquidity, which needs to be provided by someone else.
```

   alt-text = "Graphic depicting where LSP services are offered"
   width = 800
   height = 348
%}

## What are Lightning wallet servers?

Once a user is connected to the lightning network, likely through an LSP, lightning wallet servers (LWS) offer services that make using lightning easier for users. LSPs sometimes offer LWS services as part of their offering but this is not exclusive to LSPs.

LWSs act as trust-minimized third-parties, so when possible users should be able to opt-out and operate their lightning wallet manually.

Below are some LWS services and the user friction points they solve.

{% include picture.html
   image = "/assets/images/guide/how-it-works/lightning-services/what-are-lightning-wallet-servers.png"
   retina = "/assets/images/guide/how-it-works/lightning-services/what-are-lightning-wallet-servers@2x.png"
   mobile = "/assets/images/guide/how-it-works/lightning-services/what-are-lightning-wallet-servers-mobile.png"
   mobileRetina = "/assets/images/guide/how-it-works/lightning-services/what-are-lightning-wallet-servers-mobile@2x.png"
   alt-text = "Graphic depicting what an LWS offers users"
   width = 800
   height = 348

> %}

Reviewer 2: Recommended text:

- User - Finds it hard to connect to lightning
- LSP - Makes it easy to connect to lightning
- LWS - Once on lightning, helps solve other usability issues

The current phrasing makes it sound like the LWS enables _the users to solve their own usability issues_. I don't think that's the case. What we really need to convey is that the LWS enables product designers and developers to solve these usability issues on behalf of users. So just simply to "helps solve other usability issues".

   mobileRetina = "/assets/images/guide/how-it-works/lightning-services/Lightning-address-mobile@2x.png"
   alt-text = "Examples of lightning addresses"
   width = 800
   height = 348
%}

### Receive payments offline

Non-custodial lightning wallets need to be online to receive payments. [Daily spending wallets]({{ '/guide/daily-spending-wallet/' | relative_url }}), in particular, are regularly offline, so receiving payments can be difficult.

LWSs can hold onto a payment (without assuming custody) and forward it to the user once they come online. A product that offers this for users is [Greenlight](https://blockstream.com/lightning/greenlight/).

This service is usually required to be offered alongside an LSP. Read more about this [here](https://medium.com/breez-technology/get-ready-for-a-fresh-breez-multiple-apps-one-node-optimal-ux-519c4daf2536).

{% include picture.html
   image = "/assets/images/guide/how-it-works/lightning-services/Offline-payment.png"
   retina = "/assets/images/guide/how-it-works/lightning-services/Offline-payment@2x.png"
   mobile = "/assets/images/guide/how-it-works/lightning-services/Offline-payment-mobile.png"

>    mobileRetina = "/assets/images/guide/how-it-works/lightning-services/Offline-payment-mobile@2x.png"

Reviewer 2: I'm having trouble parsing one of the sentences in this image. "Is incoming from the lightning network hits the LWS".

Maybe _"Is incoming from the lightning network and hits the LWS"._

"And" being the key word that makes the sentence easier to read.

Author: Done :)

> LWS can act as third parties that non-custodially hold onto a payment and forward it to the user once they come online. A product that offers this for users is [Greenlight](https://blockstream.com/lightning/greenlight/).

Reviewer 2: ```suggestion
LWSs can hold onto a payment (without assuming custody) and forward it to the user once they come online. A product that offers this for users is [Greenlight](https://blockstream.com/lightning/greenlight/).
```

>    alt-text = "How an LWS helps users accept payments whilst offline"

Reviewer 2: ```suggestion
   alt-text = "How an LWS helps users accept payments while offline"
```
Unnecessarily hipster

   width = 800
   height = 348
%}

### Swaps

Moving bitcoin between lightning and on-chain can be costly and complicated if done by opening and closing payment channels.

LWSs can offer [submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/) to allow users to easily move bitcoin between lightning and on-chain without having to manage and deal with payment channels.

{% include picture.html
   image = "/assets/images/guide/how-it-works/lightning-services/Submarine-swaps.png"
   retina = "/assets/images/guide/how-it-works/lightning-services/Submarine-swaps@2x.png"
   mobile = "/assets/images/guide/how-it-works/lightning-services/Submarine-swaps-mobile.png"
   mobileRetina = "/assets/images/guide/how-it-works/lightning-services/Submarine-swaps-mobile@2x.png"
   alt-text = "How an LWS can conduct swaps moving bitcoin between on-chain and lightning"
   width = 800
   height = 348

> %}

Reviewer 2: Need some capital Bs in these images because every other title sentence fragment is capitalized at the beginning.

- ~"bitcoin on lightning"~ -> "Bitcoin on lightning"
- ~"bitcoin on-chain"~ => "Bitcoin on-chain"

> %}

Reviewer 2: While we agreed to lowercase bitcoin, it needs to be capitalized in this context because it's the beginning of a sentence (or title)

### Watchtowers

> Users need to regularly be online to monitor their payment channels to prevent cheating attempts. [Daily spending wallets]({{ '/guide/daily-spending-wallet/' | relative_url }}) in particular are regularly offline and so cheating attempts are more likely.
> 
> A [watchtower](https://bitcoinops.org/en/topics/watchtowers/) LWS service monitors payment channels of users who are offline and can punish their counterparties if they attempt to steal the users funds.

Reviewer 2: ```suggestion
A lightning wallet must be online on a regular basis to track it's payment channels. A [daily spending wallet]({{ '/guide/daily-spending-wallet/' | relative_url }}), in particular, is offline whenever the user stops using the app. This means there is a greater risk of cheating attempts.

A [watchtower](https://bitcoinops.org/en/topics/watchtowers/) service can fix this. Watchtowers monitor the payment channels of offline users. If a counterparty attempts to steal a user's funds, the watchtower can step in to help. The watchtower can prevent the theft by submitting a justice transaction.
```

A couple of things.
- In sentence 1, we talk about a "user" being offline, but in sentence 2 we talk about a "wallet" being offline. Let's keep it consistent throughout the paragraph.
  - IMO, better to talk about the wallet being offline.
  - Example: I don't really care if the user is looking at the app or not -- if Apple and Google decided that bitcoin wallets could conduct lightning activity in the background, that would basically solve a lot of these issues. So the issue itself is not necessarily about the user being online but rather the node (the app) being online.
- Changed "punishment" to "justice transaction" in futile attempt to thwart BDSM jokes before they arrive.
- Simplified language a good bit
- Do we need to end words like "watchtower" with "LWS"? I feel like the words are more easily understood if we strip out jargon. Already have to teach the reader what a watchtower is, calling it a "watchtower LWS service" just makes it sound more complicated.

{% include tip/open.html color="blue" icon="info" label="Using watchtowers" %}

> A watchtower should be a different user than the one the user has channels with, such as an LSP, as they have nothing at stake to attempt to cheat the user.

Reviewer 2: ```suggestion
It's best practice to use a watchtower that is not controlled by the user's LSP. This way, the watchtower has no incentive to cheat the user.
```
Calling the watchtower a "user" brings up the wrong mental image for me. A watchtower could be another user, or it could be more like a service.

---
layout: guide
title: Lightning Services
description: Third party services that solve common user experience problems when connecting to and using the lightning network.
nav_order: 4
parent: How it works
permalink: /guide/how-it-works/lightning-services/
redirect_from:
 - /guide/payments/send/lightning-service-providers/
main_classes: -no-top-padding
---

<!--

Editor's notes

Illustration sources: https://www.figma.com/file/qr4P17z6WSPADm6oW0cKw2/?node-id=701%3A5911

-->

{% include picture.html
   image = "/assets/images/guide/how-it-works/lightning-services/lightning-services.jpg"
   retina = "/assets/images/guide/how-it-works/lightning-services/lightning-services@2x.jpg"
   mobile = "/assets/images/guide/how-it-works/lightning-services/lightning-services-mobile.jpg"
   mobileRetina = "/assets/images/guide/how-it-works/lightning-services/lightning-services-mobile@2x.jpg"

>    alt-text = "Lightning service providers header image."

Reviewer 3: Alt-text is not useful. If the image does not provide important information for the reader, you can leave the alt-text blank.

   alt-text = "Graphic depicting how LSPs offer liquidity as a service"
   width = 800
   height = 348
%}

## How do LSPs connect users to lightning?

To use lightning, a self-custodial user needs at least one  [payment channel]({{ '/guide/getting-started/technology-primer/#what-is-a-payment-channel' | relative_url }}). To send payments, this channel needs outbound liquidity, which the user usually provides. And to receive payments, this channel needs inbound liquidity, which needs to be provided by someone else.

Finding someone offering inbound liquidity and opening a channel with them can be difficult, especially for new users. LSPs provide inbound liquidity and open channels for users.

{% include picture.html
   image = "/assets/images/guide/how-it-works/lightning-services/How-is-service-offered.png"
   retina = "/assets/images/guide/how-it-works/lightning-services/How-is-service-offered@2x.png"
   mobile = "/assets/images/guide/how-it-works/lightning-services/How-is-service-offered-mobile.png"
   mobileRetina = "/assets/images/guide/how-it-works/lightning-services/How-is-service-offered-mobile@2x.png"

>    alt-text = "Graphic depicting how LSPs offer inbound liquidity as a service to users"

Reviewer 3: Alt-texts should include not phrases like "Graphic of..." ([tips on alt text](https://www.a11yproject.com/posts/alt-text/)).
